src/tests/readytorun/crossgen2/Program.cs

```diff
@@ -1537,7 +1542,9 @@ private unsafe static bool ExplicitlySizedStructTest()
                 return false;
         }
 
+        if (sizeof(IntPtr) == 8) // This test isn't right for 32 bit platforms
```

Comment:
Hmm, I'm somewhat confused here - you're disabling the test on x86 yet you changed the <code>Size</code> of the <code>StructLayout</code> attribute on <code>ExplicitlySizedStructAuto</code> - does that mean that the previous value was wrong or is the change from 0x14 to 0x18 supposed to be more stressful? I don't think it's a big deal, it just comes as a surprise why you need to modify a test you're disabling for the "new x86 scenario" you're adding.

Reply:
Well, the test is just wrong. It turns out it doesn't pass when run without crossgen2, with exactly the same fault, but by putting the disable logic as IL in the test, we actually get testing that the behavior of the x86 type loader at runtime matches up with the behavior at crossgen2 time due to the presence of the --verify-type-and-field-layout switch that we use when compiling crossgen2 tests.

I agree though, I was quite disappointed to find that I needed to change the test.

As far as how much was passing, as I mentioned to you a couple days ago, I was looking for something different than endless infrastructure stuff, so I took a look at this, and its come pretty clean. At this point I believe our type layout is very nearly correct, as is our argument layout handling, which leaves whatever interesting details are special about x86 in the jit interface. I have some suspicions that then handling of single element struct normalization isn't quite right, but its not all that far off from right.